### ISSUE TITLE
State of Maintenance

### ISSUE THREAD

**Issue by nateprewitt** [postmanlabs/httpbin#702]
Hi there :wave:

I'm reaching out regarding the maintenance of httpbin. There have been a couple outstanding PRs to help move httpbin forward with changes around Flask and its underlying components. Specifically that there's currently no working CI to test changes and #684 to help enable launching Python 3.12 support.

I wanted to check on the current maintenance status for the project and if there's intention to continue its development. Is there a good point of contact to help get the repo cleaned up or feature requests reviewed?

**Comment by kevin1024** [comment]
```*an ancient maintainer emerges from hibernation*```

Hello there!

Well, I seem to have been removed as a maintainer on the this repo, but I am still a maintainer on pypi.  Perhaps I can help with a new release!

Shall we fork the repo?

**Comment by mgorny** [comment]
Yes, please. Patches and workarounds are now piling up in Gentoo. I'm happy to help getting things merged and moving them forward once there's a single place for them.

**Comment by ewdurbin** [comment]
If postman folks are amenable, I'd support taking this on in the @psf org, and possibly hosting for httpbin.org itself in the PSF infrastructure. (Though I would be interested to know what the current traffic numbers look like first).

**Comment by ewdurbin** [comment]
I've raised this at https://community.postman.com/t/status-of-maintenance-for-httpbin/48649 to try to get some response.

**Comment by ewdurbin** [comment]
No response on discussion forum. I went ahead and sent a support request 🤷🏼.

**Comment by 4cq2** [comment]
https://github.com/postmanlabs/httpbin/issues/703

**Comment by ghost** [comment]
I am getting random time-outs now:

~~~
> curl -i httpbin.org/get
HTTP/1.1 504 Gateway Time-out
Server: awselb/2.0
Date: Sat, 24 Jun 2023 15:43:02 GMT
Content-Type: text/html
Content-Length: 132
Connection: keep-alive

<html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h1>504 Gateway Time-out</h1></center>
</body>
</html>
~~~

I would recommend people switch to:

https://postman-echo.com

**Comment by mgorny** [comment]
The problem is not really the online instance but the package that (unlike online instances) can be used to test stuff offline.

**Comment by ewdurbin** [comment]
Still no response from postman folks. @kevin1024 @nateprewitt are you interested in forking at this point? I'd be happy to setup such a fork in the @psf organization, mapping the current owners on PyPI to the admins of the project.

**Comment by kevin1024** [comment]
I’m still game for a fork.

**Comment by nateprewitt** [comment]
Yep, let's do it.

**Comment by ewdurbin** [comment]
Will proceed!

**Comment by ewdurbin** [comment]
Invites dispatched to the org/team for @nateprewitt, @kevin1024, @sigmavirus24, and @timofurrer.

**Comment by ewdurbin** [comment]
https://github.com/psf/httpbin now exists as a fork. Once accepted, the names above will have Maintainer privileges on the repository. We can have a discussion there over Admin level privileges.

**Comment by exhuma** [comment]
What will happen with the existing issues? I am currently looking into replacing a home-grown solution with `httpbin`. Unfortunately it has to run on OpenShift which does not allow to use privileged ports (`80` in this case). I could open a PR for that. The change to allow custom ports via env-var is fairly minimal.

I would like to open an Issue about this. But the fork currently has issues disabled. Should I open it here instead? Or wait until they are available in the fork?

**Comment by ewdurbin** [comment]
Thanks for flagging. I’ve enabled Issues on the fork.

**Comment by kevin1024** [comment]
Well, I merged a couple of PRs that should fix the major issues with werkzeug 3 incompatibility and packaging over on the branch and have pushed a new version of httpbin to pypi.  I hope this doesn't break too many builds out there and is a net positive for the world of open source http libraries :)

**Comment by nateprewitt** [comment]
We're strictly pinned for Requests currently but I'll take a look at testing the new release soon. Thanks @kevin1024!

**Comment by DanranWf** [comment]
unsubscribe













--

祝好！
连文静




在 2023-07-10 23:27:44，"Ee Durbin" ***@***.***> 写道：

Will proceed!

**Comment by immanuelfodor** [comment]
Actually, the project offers a nice Swagger/OpenAPI spec but it's invalid on multiple aspects (eg. uses old SmartAPI fields).
Could we get this small PR merged for the new release to fix these? https://github.com/postmanlabs/httpbin/pull/710

**Comment by sigmavirus24** [comment]
@immanuelfodor please send that to the psf fork to get it merged

**Comment by immanuelfodor** [comment]
Thanks for the suggestion! Where is the pfs fork deployed, in which domain will the change be live once merged?

**Comment by exhuma** [comment]
The repo is here: https://github.com/psf/httpbin
I just noticed that I can't see the container image. Maybe just a permission issue. I'll ping one of the maintainers. The project recently moved over to that repo.

**Comment by immanuelfodor** [comment]
Yes, I've found it, but my question was more about if the psf repo is the one that's deployed to https://httpbin.org/ now instead of the postmanlabs repo? Or if not, is there a "live" version of the psf one somewhere? Or it's just a Docker-only release for psf?

BTW thanks for checking the Docker release!

**Comment by exhuma** [comment]
Unfortunately I don't know the answer to this. I *assume* that the domain is still owned by K. Reitz so chances are - currently - small that it will be deployed there.
